src/test/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenterTest.java

```
@@ -0,0 +1,1190 @@
+     * extracted by the AlkylStructureFragmenter - an isolated double bond, a quaternary C, an aromatic ring system with
+     * a connected conjugated system, an isolated cyclohexane ring and a tertiary C - in default settings.
+     *
+     * @throws CDKException if SMILES cannot be parsed correctly
+     */
+    @Disabled
+    @Test
+    public void extractFragmentsTest() throws CDKException {
+        ArrayList<String> tmpExpectedFragmentsList = new ArrayList<>(6);
+        tmpExpectedFragmentsList.add("C=CC1=CC=C2C=CC=CC2=C1");
+        tmpExpectedFragmentsList.add("CC(C)(C)C");
+        tmpExpectedFragmentsList.add("C=C");
+        tmpExpectedFragmentsList.add("CCC");
+        tmpExpectedFragmentsList.add("CCC");
+        tmpExpectedFragmentsList.add("C1CCCCC1");
+        SmilesParser tmpParser = new SmilesParser(SilentChemObjectBuilder.getInstance());
+        IAtomContainer tmpTestAC = tmpParser.parseSmiles("C12=CC(=CC=C1C(=CC(=C2)C3CCCCC3)CC(C)(C)C/C=C/CCC)/C=C\\C(C)C");
+        MolecularArrays tmpTestMolecularArrays = new MolecularArrays(tmpTestAC);
+        AlkylStructureFragmenter tmpASF = new AlkylStructureFragmenter();
+        tmpASF.markTertQuatAndNeighbors(tmpTestMolecularArrays);
+        tmpASF.markRings(tmpTestMolecularArrays, tmpTestAC);
+        tmpASF.markConjugatedPiSystems(tmpTestMolecularArrays);
+        tmpASF.markMultiBonds(tmpTestMolecularArrays);
+        IAtomContainerSet tmpExtractedFrag;
+        ArrayList<String> tmpExtractedFragmentList = new ArrayList<>(6);
+        try {
+            tmpExtractedFrag = tmpASF.extractFragments(tmpTestMolecularArrays);
+            for (IAtomContainer tmpAC : tmpExtractedFrag.atomContainers()) {
+                ChemUtil.saturateWithHydrogen(tmpAC);
+                tmpExtractedFragmentList.add(ChemUtil.createUniqueSmiles(tmpAC, false));
+            }
+        } catch (CloneNotSupportedException | CDKException e) {
+            Assertions.fail();
```

In tests, you can simply let the exception be thrown, no need for try-catch.